session/auth/authprovider/authprovider.go

```diff
@@ -89,6 +95,12 @@ func (ap *authProvider) FetchToken(ctx context.Context, req *auth.FetchTokenRequ
 		Secret:   creds.Secret,
 	}
 
+	var httpClient = http.DefaultClient
+	if tc, err := ap.tlsConfig(req.Host); err == nil {
+		httpClient.Transport = http.DefaultTransport
+		httpClient.Transport.(*http.Transport).TLSClientConfig = tc
```

Comment:
🤔 I don't think we should be modifying the `http.DefaultTransport` instance.

Reply:
yes, you are right.